src/later.cpp

```diff
@@ -155,5 +200,6 @@ double next_op_secs() {
 
 extern "C" void execLaterNative(void (*func)(void*), void* data, double delaySecs) {
   ensureInitialized();
-  doExecLater(func, data, delaySecs);
+  int loop = GLOBAL_LOOP;
+  doExecLater(getCallbackRegistry(GLOBAL_LOOP), func, data, delaySecs, loop == GLOBAL_LOOP);
```

Comment:
Can't call getCallbackRegistry here, it's ASSERT_MAIN_THREAD. I think we're going to have to guard the callbackRegistries somehow.

Reply:
Also it'd be great if we had a unit test that identified this issue.

Reply:
Two things:

* We need to protect accesses to the `callbackRegistries` to be thread safe.
* We need to figure out what to do if a background thread tries to schedule a callback on the main thread, but with a loop that no longer exists.
* Do we need lock hierarchies?